src/native-state-drm.cpp

```diff
@@ -212,6 +212,25 @@ udev_drm_test_primary_gpu(UDEV_TEST_FUNC_SIGNATURE(, current_device,))
     return is_main_gpu;
 }
 
+/* Test if the drm-device is actually modeset capable.
+ * Render-only devices cannot drive an actual display,
+ * so the GETRESOURCES ioctl will fail in that case.
+ */
+static bool udev_drm_test_modeset(std::string dev_path)
+{
+    struct drm_mode_card_res res;
```

Comment:
We need to zero-initialize the struct (or at least the res.count_* members), otherwise the kernel will start writing info to the (invalid) res.*_ptr memory locations. The following should do the trick:

```suggestion
    struct drm_mode_card_res res{};
```